packages/react-hooks/src/RecordProvider.tsx

```diff
@@ -0,0 +1,69 @@
+import type { BaseRecord, RecordSavedEvent, RecordDeletedEvent, RecordUpdatedEvent } from '@aries-framework/core'
+
+import { RepositoryEventTypes } from '@aries-framework/core'
+import { useReducer } from 'react'
+
+export enum RecordProviderEventTypes {
+  RecordsLoaded = 'RecordsLoaded',
+}
+
+export interface RecordsState<R extends BaseRecord<any, any, any>> {
```

Comment:
There's quite some eslint warnings in this file. In the redux package I created the following type with a single eslint disable:

```
// eslint-disable-next-line @typescript-eslint/no-explicit-any
type BaseRecordAny = BaseRecord<any, any, any>
```